prow/git/v2/client_factory.go

```diff
@@ -387,16 +387,19 @@ func (c *clientFactory) ClientForWithRepoOpts(org, repo string, repoOpts RepoOpt
 
 	// First create or update the primary clone (in "cacheDir").
 	timeBeforeEnsureFreshPrimary := time.Now()
-	c.ensureFreshPrimary(cacheDir, cacheClientCacher, repoOpts, org, repo)
-	gitMetrics.ensureFreshPrimaryDuration.WithLabelValues(org, repo).Observe((float64(time.Since(timeBeforeEnsureFreshPrimary).Seconds())))
+	err = c.ensureFreshPrimary(cacheDir, cacheClientCacher, repoOpts, org, repo)
+	if err != nil {
+		c.logger.WithFields(logrus.Fields{"org": org, "repo": repo, "dir": cacheDir}).Errorf("Error encountered while refreshing primary clone: %s", err.Error())
+	}
+	gitMetrics.ensureFreshPrimaryDuration.WithLabelValues(org, repo).Observe(time.Since(timeBeforeEnsureFreshPrimary).Seconds())
```

Comment:
This metric should only be measured if the operation succeeds.

Reply:
I think we still want to know the duration even if the git operations failed. This is also the previous behavior when the error was not handled.

Reply:
Then you should add another field for "success" so that we don't pollute the metrics.

If we were combining durations of failures and successes together, that was probably unintentional and should be fixed (as above), or by not measuring when we fail.

Reply:
Ok, changed to observe only when succeeded.